contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -317,16 +325,25 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
 
     bool success;
     uint256 gasUsed;
+    // allocate return data memory ahead of time
+    bytes memory returnData = new bytes(MAX_CALLBACK_RETURN_BYTES);
 
     // solhint-disable-next-line no-inline-assembly
     assembly {
+      // solidity calls check that a contract actually exists at the destination, so we do the same
+      // Note we do this check prior to measuring gas so gasForCallExactCheck (our "cushion")
+      // doesn't need to account for it.
+      if iszero(extcodesize(client)) {
+        revert(0, 0)
+      }
+
       let g := gas()
-      // GAS_FOR_CALL_EXACT_CHECK = 5000
-      // Compute g -= GAS_FOR_CALL_EXACT_CHECK and check for underflow
-      // The gas actually passed to the callee is min(gasAmount, 63//64*gas available).
+      // GASFORCALLEXACTCHECK = 5000
+      // Compute g -= gasForCallExactCheck and check for underflow
+      // The gas actually passed to the callee is _min(gasAmount, 63//64*gas available).
```

Comment:
Why is it 63/64 * gasAvailable?

Reply:
what is gasAmount? Is this `g`?
ie:
gasAvailable = gas()
gasAmount = gas() - GAS_FOR_CALL_EXACT_CHECK?

The variable names in the code don't match the variable names in the comments which makes it confusing.

Reply:
I know it is outside of the scope of change for this PR, but just making a note. Looks like that 1/64 rule is pretty complex, so I am hoping this code was implemented by someone who had a solid understanding.

Reply:
This is lifted from CCIP, which just went through audit. So it is, yeah.